mlflow/tracking/client.py

```diff
@@ -1536,16 +1555,45 @@ def log_image(
                 "`key` to log dynamic image charts or `artifact_file` for saving static images. "
             )
 
+        import numpy as np
+
+        # Convert image type to PIL if its a numpy array
+        if isinstance(image, np.ndarray):
+            image = _convert_numpy_to_pil_image(image)
+        elif isinstance(image, Image):
+            image = image.to_pil()
+
         if artifact_file is not None:
-            self._log_image_as_artifact(run_id, image, artifact_file)
+            with self._log_artifact_helper(run_id, artifact_file) as tmp_path:
+                image.save(tmp_path)
 
         elif key is not None:
+            # Check image key for invalid characters
+            if not re.match(r"^[a-zA-Z0-9_\-./ ]+$", key):
```

Comment:
do we have other requirements like `-` cannot be the leading character?

Reply:
I'll look more into it, but `-` being the leading character seems to work fine